api/ocm/resolvers/component.go

```diff
@@ -0,0 +1,137 @@
+package resolvers
```

Comment:
shouldn't it be possible to define such a component resolver based on similar rules as the component version resolver through a configuration object, so it also can be configured through the command line and retrieved from the context?

Reply:
The component resolver is a special case required for particular application code.
In the library the only ComponentResolver is the MatchingResolver configured for the OCM Context.
This one is used by some special ref handlers in the CLI (request from Christian).

But this resolver is already configurable by the config management.

The only other application, so far, requiring a component resolver is the service model, because we have to resolve version constraints, here. This requires a ListVersions which is only available for a component. Therefore, a ComponentVersionResolver cannot be used.

The CLI plugin for the ServiceModel would use the context resolver, here, which is already configurable.